cluster/src/test/scala/org/apache/spark/sql/execution/SnappyTableMutableAPISuite.scala

```diff
@@ -345,24 +345,43 @@ class SnappyTableMutableAPISuite extends SnappyFunSuite with Logging with Before
     assert(resultdf.contains(Row(88, 88, 90)))
   }
 
+  test("PutInto Cache Test") {
+    val snc = new SnappySession(sc)
+    val rdd = sc.parallelize(data1, 2).map(s => Data(s(0), s(1), s(2)))
+    val df1 = snc.createDataFrame(rdd)
+    val rdd2 = sc.parallelize(data2, 2).map(s => Data(s(0), s(1), s(2)))
+    val df2 = snc.createDataFrame(rdd2)
+    snc.sql("set spark.sql.defaultSizeInBytes=1000")
+    snc.createTable("col_table", "column", df1.schema, Map("key_columns" -> "col2"))
+
+    df1.write.insertInto("col_table") // insert initial data
+    df2.write.putInto("col_table") // update & insert subsequent data
+    df2.write.putInto("col_table")
+    val resultdf = snc.table("col_table").collect()
+    assert(resultdf.length == 7)
+    assert(resultdf.contains(Row(8, 8, 8)))
+    assert(resultdf.contains(Row(88, 88, 90)))
+  }
+
   test("PutInto with API for pure inserts") {
     val snc = new SnappySession(sc)
     val rdd = sc.parallelize(data1, 2).map(s => Data(s(0), s(1), s(2)))
     val df1 = snc.createDataFrame(rdd)
     val df2 = snc.range(100, 110).selectExpr("id", "id", "id")
-
-    snc.createTable("col_table", "column", df1.schema, Map("key_columns" -> "col2"))
+    snc.sql("set spark.sql.defaultSizeInBytes=1000")
```

Comment:
What does "defaultSizeInBytes" do?

Reply:
This sets the size of a RDD if there is no reasonable statistics about its statistics.